### DISCUSSION TITLE
AR viewer on Android mobile not opening

### DISCUSSION THREAD

**Discussion by superLeaAtWork** [google/model-viewer#4025] [Q&A]
Hello everybody, 

we integrated the model viewer into our website. The model viewer displays the model correctly but it can be placed in the room using the AR viewer only on MOST devices. It works on all IOS devices as far as we tested and on the most Android devices. 

However, when I try it with my phone and one of my colleagues, the AR viewer tries to open, but closes itself immediately. Sometimes I get the notification that Google gets closed several times. 

It is strange because when I integrate the model viewer into a simple page it works. 

#### Live Demo

https://glitch.com/edit/#!/model-viewer

Here some links where the **AR viewer WORKS** on my device: 
https://diva3d.diva-portal.com/test3.html
https://diva3d.diva-portal.com/testios.html
https://diva3d.diva-portal.com/test.html
The astronaut works here: https://modelviewer.dev/
On this page all but one viewer do work: https://modelviewer.dev/examples/augmentedreality/index.html

Here some links where the **AR viewer DOES NOT WORK** on my device: 

- When I try to open the AR viewer with the astronaut model that is integrated using the following code: https://modelviewer.dev/examples/augmentedreality/index.html

<model-viewer src="../../shared-assets/models/Astronaut.glb" ar ar-scale="fixed" camera-controls alt="A 3D model of an astronaut" skybox-image="../../shared-assets/environments/aircraft_workshop_01_1k.hdr" ios-src="../../shared-assets/models/Astronaut.usdz" xr-environment></model-viewer>
 Its interesting because https://diva3d.diva-portal.com/test3.html uses the same settings and there it works. 

 - When trying to using the AR functionality of this chair: https://www.pfister.ch/de/produkt/2203707/byyu-armlehnstuhl-sergio-3729

### Browser Affected

I tested it on Chrome and Firefox: 
- [ ] Chrome, version: 100.0.4896.127
- [ ] Firefox: 99.2.0

### OS

- Android where it's not working: 
Samsung Galaxy S9 SM-G960F
Android Version 10 
Google Version: 13.14.9.23.arm64

- Android where it's not working: 
Redmi Note 9 Modell M2003J15SG
Android Version 10 QP1A.190711.020
Google Version: 13.14.9.23.arm64

- Android where IT WORKS: 
Redmi Note 9 Pro
Android Version 11 RKQ1.200826.002

Is it possible that it does not work correctly on phones with Android version 10?

I hope you can help me with this issue. We already tested a lot and have no idea what we can do next :) 

Thanks in advance!

Lea

**Comment by elalish** [comment]
Thanks, it does sound like an Android 10 thing, but that ought to work. Can you restrict `ar-modes` to just `scene-viewer` and then just `webxr` and let us know of any difference in behavior? Also, please open your JS console and paste any warnings you get there.

**Comment by superLeaAtWork** [comment]
Hello, it seams that the problem has solved itself after updating to the newest 1.12 version!

**Comment by superLeaAtWork** [comment]
Hello everybody, 

I fear the issue is still not solved. 
After updating to 1.12 it worked when starting the application in my dev environment and connecting that one with my smartphone. 

However, after building everything I saw that the AR viewer still does not work on my phone. 

Then, I connected my phone with debugging mode to my PC and found an interesting information. We use 'webxr scene-viewer quick-look' as arModes.
In my localhost the application tries to start AR with the scene-viewer. 
However, in the deployed version, it tries to start AR with webXR. 

I now tested the following arModes: 'webxr quick-look'; Even in my dev environment my phone does not even print the AR button to place the model in the room. 
When using following configuration 'scene-viewer quick-look' everything works fine. 
There is no error in the console in both cases. 

So I think the problem is quite obvious. When starting AR in the build application with following configuration 'webxr scene-viewer quick-look' , my phone does not detect that webxr is not supported. So it does not uses the scene-viewer as fallback. 
Is this a model-viewer bug? 

Regards, 
Lea

**Comment by elalish** [comment]
Sounds more like a WebXR bug to me, as we use their API to determine support. @klausw Can you repro? Android 10 is a little old, but we still test there, right?

**Comment by elalish** [comment]
@superLeaAtWork One requirement of WebXR is that it's on HTTPS. Might that be the problem? There are also extra permissions needed if this is in an iframe.

**Comment by superLeaAtWork** [comment]
Hello @elalish , 
I don't think so. The webseite we are integrating it into also uses https. 
Regards, 
Lea

**Comment by klausw** [comment]
Android 10 should be OK - I just tried a Samsung Galaxy S8+ running the rather ancient Android 7.0, and the following work for me:

* the leather chair on https://modelviewer.dev/examples/augmentedreality/index.html 
* the astronaut on https://modelviewer.dev/examples/augmentedreality/index.html (I confirmed in Inspector it's using WebXR)

The white chair model on https://www.pfister.ch/de/produkt/2203707/byyu-armlehnstuhl-sergio-3729 works, but appears to be using scene viewer, not WebXR.

@superLeaAtWork , would you mind trying the following pages from https://immersive-web.github.io/webxr-samples/ to confirm if WebXR AR mode is working in general?

* https://immersive-web.github.io/webxr-samples/immersive-ar-session.html
* https://immersive-web.github.io/webxr-samples/ar-barebones.html

**Comment by superLeaAtWork** [comment]
Hallo, 

I tested the following two links on the following two devices and both work fine: 
https://immersive-web.github.io/webxr-samples/immersive-ar-session.html
https://immersive-web.github.io/webxr-samples/ar-barebones.html

Samsung Galaxy S9 SM-G960F
Android Version 10
Google Version: 13.14.9.23.arm64

Redmi Note 9 Modell M2003J15SG
Android Version 10 QP1A.190711.020
Google Version: 13.14.9.23.arm64

Here again the link where it does not work on both devices: 
https://demo.diva-portal.com/products?lang=de&channelId=cWuuBtYPF&organizationId=ba334319-e17c-43a4-9fb8-3a67a27be19e&id=0Ap8E-FioL

Best regards, 
Lea

**Comment by superLeaAtWork** [comment]
Hello everybody, 

I just wanted to ask if there are any news on this topic? 
Is it possible to reopen this issue? 

Best regards, 
Lea

**Comment by elalish** [comment]
@klausw Can you repro?

**Comment by klausw** [comment]
When I try https://demo.diva-portal.com/products?lang=de&channelId=cWuuBtYPF&organizationId=ba334319-e17c-43a4-9fb8-3a67a27be19e&id=0Ap8E-FioL on a Samsung S8, pressing "phone in front of cube" icon appears to enter model-viewer mode, and this shows the expected "Enter AR" button at the bottom right. Pressing that shows the XR permission prompt (on first use), briefly enters AR mode, then shows a black screen with the fullscreen instructions on top. It then immediately returns back to 2D mode.

If I'm understanding it right, your issue was that model-viewer's "Enter AR" button was missing due to apparently not detecting `immersive-ar` mode support? If that's the case, I can't reproduce it.

There are various errors on the JS console:

Attempting to present in AR with WebXR...
products:1 Access to fetch at 'https://images.diva-portal.com/MESSE21/SPINS/ZIL/DIVA-527771-1/DIVA-527771-1.spin?scale.option=noup&w=3000&info=sirv_spin_info_v1_1400620957' from origin 'https://demo.diva-portal.com' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: It does not have HTTP ok status.
7552.0104f09d.js:2          GET https://images.diva-portal.com/MESSE21/SPINS/ZIL/DIVA-527771-1/DIVA-527771-1.spin?scale.option=noup&w=3000&info=sirv_spin_info_v1_1400620957 net::ERR_FAILED

XHR error. Switching to JSONP.
sirv.js:8          GET https://images.diva-portal.com/MESSE21/SPINS/ZIL/DIVA-527771-1/DIVA-527771-1.spin?scale.option=noup&w=3000&info=sirv_spin_info_v1_1400620957&callback=sirv_spin_info_v1_1400620957 net::ERR_ABORTED 400

products:1 Access to fetch at 'https://images.diva-portal.com/MESSE21/SPINS/ZIL/DIVA-527771-1/DIVA-527771-1.spin?scale.option=noup&w=3000&info=sirv_spin_info_v1_1400620957' from origin 'https://demo.diva-portal.com' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: It does not have HTTP ok status.
7552.0104f09d.js:2          GET https://images.diva-portal.com/MESSE21/SPINS/ZIL/DIVA-527771-1/DIVA-527771-1.spin?scale.option=noup&w=3000&info=sirv_spin_info_v1_1400620957 net::ERR_FAILED
XHR error. Switching to JSONP.
sirv.js:8          GET https://images.diva-portal.com/MESSE21/SPINS/ZIL/DIVA-527771-1/DIVA-527771-1.spin?scale.option=noup&w=3000&info=sirv_spin_info_v1_1400620957&callback=sirv_spin_info_v1_1400620957 net::ERR_ABORTED 400

**Comment by superLeaAtWork** [comment]
Hello, 

no the "Enter AR" button appears. Only it tries to open the viewer where I can place the object with my camera in the room and closes it immediately after. 

Regards Lea

**Reply by elalish** [reply]
Closing the issue for lack of feedback. Hoping that means it's been resolved by other fixes that have shipped. Will reopen if new information arises.

**Comment by elalish** [comment]
This issue has gone through a few iterations now; @superLeaAtWork does this problem still repro with v2.0?
